### PR TITLE
*: add maxprocs metrics for mcs

### PR DESCRIPTION
**Pull request by CabinfeverB**
### What problem does this PR solve?

Issue Number: ref #5839

### What is changed and how does it work?
move pd_service_maxprocs into `pkg/basicserver` for mcs and pd server

```commit-message
```

### Check List

Tests

- Integration test

Code changes

- Has the configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/basicserver/metrics.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package server
+
+import "github.com/prometheus/client_golang/prometheus"
+
+var (
+	ServerMaxProcs = prometheus.NewGauge(
```
Comment (rleungx) [inline pkg/basicserver/metrics.go:20]: How about serverInfo?

**Review by rleungx — APPROVED** [review]

File: pkg/basicserver/metrics.go [hunk 2]
```diff
@@ -16,22 +16,27 @@ package server
...
+	ServerMaxProcsGauge = prometheus.NewGauge(
+		prometheus.GaugeOpts{
+			Namespace: "pd",
+			Subsystem: "service",
+			Name:      "maxprocs",
+			Help:      "The value of GOMAXPROCS.",
+		})
+
+	// ServerInfoGauge indicates the pd server info including version and git hash.
+	ServerInfoGauge = prometheus.NewGaugeVec(
```
Comment (rleungx) [inline pkg/basicserver/metrics.go:30]: Another option, we can define the metrics for different services here.

Comment (CabinfeverB) [reply pkg/basicserver/metrics.go:30]: One metrics is maybe enough. We can get instance roles from service-role plane and find version info by instance.

**Review by lhy1024 — APPROVED** [review]

**Pull request merged**